jaeger-core/src/main/java/com/uber/jaeger/reporters/protocols/JaegerThriftSpanConverter.java

```diff
@@ -47,11 +54,24 @@
             context.getFlags(),
             span.getStart(),
             span.getDuration())
+            .setReferences(buildReferences(span.getReferences()))
         .setTags(buildTags(span.getTags()))
         .setLogs(buildLogs(span.getLogs()));
   }
 
-  protected static List<Log> buildLogs(List<LogData> logs) {
+  static List<SpanRef> buildReferences(List<Reference> references) {
+    List<SpanRef> thriftReferences = new ArrayList<SpanRef>(references.size());
+    for (Reference reference: references) {
+      SpanRefType thriftRefType = References.CHILD_OF.equals(reference.getType()) ? SpanRefType.CHILD_OF :
```

Comment:
I would pass a spanContext and add this check, to reduce the side of the span message:
```
if (References.CHILD_OF.equals(reference.getType()) &&
    spanContext.getTraceID() == reference.getTraceID() &&
    spanContext.getParentId() == reference.getSpanID()) {
    continue; // skip child_of reference that's already represented by parentID
}
```